BrainPortal/app/helpers/userfiles_helper.rb

```diff
@@ -77,16 +77,24 @@ def file_link_table(previous_userfile, next_userfile, sort_index, options = {})
   end
 
   # Generates links to pretty file content for userfiles
-  # of type TextFile or ImageFile; this method is going to
-  # be replaced by a proper generic framework in 4.3.0 !
+  # of type TextFile or ImageFile
+  # Generates download link for any other type of file
   def data_link(file_name, userfile)
-    display_name  = Pathname.new(file_name).basename.to_s
+    dp = DataProvider.find(userfile.data_provider_id)
+    cache_path = dp.cache_full_path(userfile)
+
+    full_path_name  = Pathname.new(cache_path.dirname + file_name)
+
+    file_lstat = full_path_name.lstat  # lstat doesn't follow symlinks, so we can tell if it is one
+
+    display_name  = full_path_name.basename.to_s
     return h(display_name) unless userfile.is_locally_synced?
+    return h(display_name) unless file_lstat.file?
```

Comment:
If possible, let's try to have the method return the default values as early as possible withotu accessing the file system if it can avoid it